public/app/features/profile/UserProfileEditPage.tsx

```diff
@@ -55,19 +63,92 @@ export function UserProfileEditPage({
   changeUserOrg,
   updateUserProfile,
 }: Props) {
+  const [queryParams, updateQueryParams] = useQueryParams();
+  const tabQueryParam = queryParams[TAB_QUERY_PARAM];
+  const [activeTab, setActiveTab] = useState<string>(
+    (typeof tabQueryParam === 'string' ? tabQueryParam : CORE_SETTINGS_TAB).toLowerCase()
+  );
+
   useMount(() => initUserProfilePage());
 
+  const extensionComponents = useMemo(() => {
+    const { extensions } = getPluginComponentExtensions({
+      extensionPointId: PluginExtensionPoints.UserProfileTab,
+      context: {},
+    });
+
+    return extensions;
+  }, []);
+
+  const groupedExtensionComponents = extensionComponents.reduce<Record<string, PluginExtensionComponent[]>>(
+    (acc, extension) => {
+      const { title } = extension;
+      if (acc[title]) {
+        acc[title].push(extension);
+      } else {
+        acc[title] = [extension];
+      }
+      return acc;
+    },
+    {}
+  );
+
+  const showTabs = extensionComponents.length > 0;
+  const tabs = [CORE_SETTINGS_TAB, ...keys(groupedExtensionComponents).map((title) => title)];
+
+  const UserProfile = () => (
+    <VerticalGroup spacing="md">
+      <UserProfileEditForm updateProfile={updateUserProfile} isSavingUser={isUpdating} user={user} />
+      <SharedPreferences resourceUri="user" preferenceType="user" />
+      <UserTeams isLoading={teamsAreLoading} teams={teams} />
+      <UserOrganizations isLoading={orgsAreLoading} setUserOrg={changeUserOrg} orgs={orgs} user={user} />
+      <UserSessions isLoading={sessionsAreLoading} revokeUserSession={revokeUserSession} sessions={sessions} />
+    </VerticalGroup>
+  );
+
+  const UserProfileWithTabs = () => (
+    <div data-testid={selectors.components.UserProfile.extensionPointTabs}>
+      <VerticalGroup spacing="md">
+        <TabsBar>
+          {tabs.map((tabTitle) => {
+            const queryParamCompatibleTabTitle = tabTitle.toLowerCase();
+
+            return (
+              <Tab
+                key={tabTitle}
+                label={tabTitle}
+                active={activeTab === queryParamCompatibleTabTitle}
+                onChangeTab={() => {
+                  setActiveTab(queryParamCompatibleTabTitle);
+                  updateQueryParams({ [TAB_QUERY_PARAM]: queryParamCompatibleTabTitle });
+                }}
+                data-testid={selectors.components.UserProfile.extensionPointTab(tabTitle)}
+              />
+            );
+          })}
+        </TabsBar>
+        <TabContent>
+          {activeTab === CORE_SETTINGS_TAB.toLowerCase() && <UserProfile />}
+          {toPairs(groupedExtensionComponents).map(([title, pluginExtensionComponents]) => {
```

Comment:
why use `toPairs` instead of `Object.entries`? 🤔 

with the new structure, this will return `[id, pluginExtensionComponents` instead of `[title, pluginExtensionComponents]`

Reply:
updated this to use `Object.entries`, thanks for the suggestion! (this is how you know I'm (mostly) a backend engineer 🤓)

this:
```tsx
Object.entries(groupedExtensionComponents).map(([title, pluginExtensionComponents])
```
will still return `[title, pluginExtensionComponents]` since I'm grouping the registered extension point components [here](https://github.com/grafana/grafana/pull/77863/files#diff-1fca821d9824581e92af5d0b7b2aa29138fdad51369ef0d93a169c3b88029c06R88-R99) by `title`. Nevertheless, I've added [this](https://github.com/grafana/grafana/pull/77863/files#diff-1fca821d9824581e92af5d0b7b2aa29138fdad51369ef0d93a169c3b88029c06R147) small util function to convert a tab title to a tab id.